pkg/nodeconfig/payload/payload.go

```diff
@@ -296,21 +300,63 @@ Compare-And-Replace-Config -ConfigPath $kubeProxyConfigPath -NewConfigContent $k
 `
 )
 
-// FileInfo contains information about a file
-type FileInfo struct {
-	Path   string
+// This is okay to be global as for each program run, this can only have one valid instance due it representing the
+// actual state of the /payload directory.
+// This should not be written to once the operator controllers have started, as it is not threadsafe to read+write at the same time.
+var shaMap map[string]string
+
+// PopulateSHAMap populates the global shaMap with the SHA256sums from the payload
+func PopulateSHAMap() error {
+	data, err := os.ReadFile("/payload/sha256sum")
+	if err != nil {
+		return fmt.Errorf("error reading from /payload/sha256sum: %w", err)
+	}
+	shaMap, err = parseShaFile(data)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// parseShaFile returns a mapping of file names to SHA256 sums
+// input should be of the format generated by the sha256sum binary: `SHA256SUM  filepath`
+func parseShaFile(fileData []byte) (map[string]string, error) {
+	m := make(map[string]string)
+	for _, line := range strings.Split(string(fileData), "\n") {
```

Comment:
I recommend to  trim spaces and check if `string(fileData)` is empty first, is so return error